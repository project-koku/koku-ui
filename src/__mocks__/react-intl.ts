const reactIntl = jest.requireActual('react-intl');

const intl = {
  formatDate: jest.fn(() => ''), // Dates won't match snapshots during PR builds
  formatDateTimeRange: jest.fn(() => ''), // Dates won't match snapshots during PR builds
  formatMessage: jest.fn(({ defaultMessage }, params?) => {
    if (!params) {
      return defaultMessage;
    }
    return defaultMessage + JSON.stringify(params);
  }),
  formatNumber: jest.fn(v => v),
};

module.exports = {
  ...reactIntl,
  createIntl: () => intl,
  defineMessages: jest.fn(v => v),
  injectIntl: jest.fn(v => v),
  useIntl: () => intl,
};
