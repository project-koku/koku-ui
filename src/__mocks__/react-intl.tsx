const mockedReactIntl = jest.genMockFromModule('react-intl') as any;

const intl = {
  formatMessage: ({ defaultMessage }) => defaultMessage,
};

mockedReactIntl.createIntl = () => intl;
mockedReactIntl.createIntlCache = () => undefined;
mockedReactIntl.defineMessages = jest.fn(v => v);
// mockedReactIntl.injectIntl = Component => props => <Component {...props} intl={intl} />;
mockedReactIntl.injectIntl = jest.fn(v => v);
mockedReactIntl.useIntl = () => intl;
mockedReactIntl.intl = intl;

module.exports = mockedReactIntl;
