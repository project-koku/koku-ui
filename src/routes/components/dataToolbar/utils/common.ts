import type { ToolbarLabelGroup } from '@patternfly/react-core';
import type { Query } from 'api/queries/query';
import { intl } from 'components/i18n';
import messages from 'locales/messages';
import { cloneDeep } from 'lodash';
import type { Filter } from 'routes/utils/filter';
import { awsCategoryKey, awsCategoryPrefix, orgUnitIdKey, tagKey, tagPrefix } from 'utils/props';

export interface Filters {
  [key: string]: Filter[] | { [key: string]: Filter[] };
}

export interface ToolbarChipGroupExt extends ToolbarLabelGroup {
  ariaLabelKey?: string;
  placeholderKey?: string;
  resourceKey?: string;
  selectClassName?: string; // A selector from routes/components/dataToolbar/dataToolbar.scss
  selectOptions?: ToolbarLabelGroup[];
}

export const defaultFilters = {
  [awsCategoryKey]: {},
  tag: {},
};

// Remove trailing commas -- see https://issues.redhat.com/browse/COST-3641
export const cleanInput = (value: string) => {
  return value.replace(/,*$/g, '').replace(/&*$/g, '');
};

export const getActiveFilters = query => {
  const filters = cloneDeep(defaultFilters);

  const parseFilters = (key, values, isExcludes = false) => {
    if (key.indexOf(tagPrefix) !== -1) {
      if (filters.tag[key.substring(tagPrefix.length)]) {
        filters.tag[key.substring(tagPrefix.length)] = [
          ...filters.tag[key.substring(tagPrefix.length)],
          ...getFilters(key, values, isExcludes),
        ];
      } else {
        filters.tag[key.substring(tagPrefix.length)] = getFilters(key, values, isExcludes);
      }
    } else if (key.indexOf(awsCategoryPrefix) !== -1) {
      if (filters[awsCategoryKey][key.substring(awsCategoryPrefix.length)]) {
        filters[awsCategoryKey][key.substring(awsCategoryPrefix.length)] = [
          ...filters[awsCategoryKey][key.substring(awsCategoryPrefix.length)],
          ...getFilters(key, values, isExcludes),
        ];
      } else {
        filters[awsCategoryKey][key.substring(awsCategoryPrefix.length)] = getFilters(key, values, isExcludes);
      }
    } else if (filters[key]) {
      filters[key] = [...filters[key], ...getFilters(key, values, isExcludes)];
    } else {
      filters[key] = getFilters(key, values, isExcludes);
    }
  };

  if (query?.filter_by) {
    Object.keys(query.filter_by).forEach(key => {
      const values = Array.isArray(query.filter_by[key]) ? [...query.filter_by[key]] : [query.filter_by[key]];
      parseFilters(key, values);
    });
  }
  if (query?.exclude) {
    Object.keys(query.exclude).forEach(key => {
      const values = Array.isArray(query.exclude[key]) ? [...query.exclude[key]] : [query.exclude[key]];
      parseFilters(key, values, true);
    });
  }
  return filters;
};

export const getChips = (filters: Filter[]): string[] => {
  const chips = [];
  if (filters instanceof Array) {
    filters.forEach(item => {
      const label = item.toString ? item.toString() : undefined;

      chips.push({
        key: item.value,
        node: label
          ? label
          : item.isExcludes
            ? intl.formatMessage(messages.excludeLabel, { value: item.value })
            : item.value,
      });
    });
  }
  return chips;
};

export const getDefaultCategory = (categoryOptions: ToolbarLabelGroup[], groupBy: string, query: Query) => {
  if (!categoryOptions) {
    return 'name';
  }
  if (query && query.group_by && query.group_by[orgUnitIdKey]) {
    return orgUnitIdKey;
  }
  for (const option of categoryOptions) {
    if (groupBy === option.key || (groupBy && groupBy.indexOf(tagPrefix) !== -1 && option.key === tagKey)) {
      return option.key;
    }
  }
  return categoryOptions[0].key;
};

export const getFilter = (
  filterType: string,
  filterValue: string,
  isExcludes = false,
  toString = undefined
): Filter => {
  return { type: filterType, value: filterValue, isExcludes, toString };
};

export const getFilters = (filterType: string, filterValues: string[], isExcludes = false): Filter[] => {
  return filterValues.map(value => getFilter(filterType, value, isExcludes));
};

export const hasFilters = (filters: Filters) => {
  if (filters) {
    for (const filterKey of Object.keys(filters)) {
      if (filterKey === awsCategoryKey || filterKey === tagKey) {
        for (const key of Object.keys(filters[filterKey])) {
          if (filters[filterKey][key]) {
            return true;
          }
        }
      } else {
        return true;
      }
    }
  }
  return false;
};

export const onDelete = (type: any, chip: any, currentFilters) => {
  let filter = null; // Clear all
  let filters = cloneDeep(defaultFilters);

  // Todo: workaround for https://github.com/patternfly/patternfly-react/issues/3552
  // This prevents us from using a localized string, if necessary
  let _type = type && type.key ? type.key : type;
  if (_type && _type.indexOf(tagPrefix) !== -1) {
    _type = _type.slice(tagPrefix.length);
  } else if (_type && _type.indexOf(awsCategoryPrefix) !== -1) {
    _type = _type.slice(awsCategoryPrefix.length);
  }

  if (_type) {
    const excludePrefix = intl.formatMessage(messages.excludeLabel, { value: '' });
    let id = chip && chip.key ? chip.key : chip;
    if (id && id.indexOf(excludePrefix) !== -1) {
      const isExcludes = id ? id.indexOf(excludePrefix) !== -1 : false;
      id = isExcludes ? id.slice(excludePrefix.length) : id;
    }
    if (currentFilters.tag[_type]) {
      filter = currentFilters.tag[_type].find(item => item.value === id);
    } else if (currentFilters[awsCategoryKey][_type]) {
      filter = currentFilters[awsCategoryKey][_type].find(item => item.value === id);
    } else if (currentFilters[_type]) {
      filter = (currentFilters[_type] as Filter[]).find(item => item.value === id);
    }

    const newFilters: any = cloneDeep(currentFilters);
    if (newFilters.tag[_type]) {
      // Todo: use ID
      newFilters.tag[_type] = newFilters.tag[_type].filter(item => item.value !== id);
    } else if (newFilters[awsCategoryKey][_type]) {
      newFilters[awsCategoryKey][_type] = newFilters[awsCategoryKey][_type].filter(item => item.value !== id);
    } else if (newFilters[_type]) {
      newFilters[_type] = newFilters[_type].filter(item => item.value !== id);
    }
    filters = newFilters;
  }
  return {
    filter,
    filters,
  };
};
