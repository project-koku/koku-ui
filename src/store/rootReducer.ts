import { notifications } from '@redhat-cloud-services/frontend-components-notifications';
import { combineReducers } from 'redux';
import { awsCostOverviewReducer, awsCostOverviewStateKey } from 'store/breakdown/costOverview/awsCostOverview';
import { azureCostOverviewReducer, azureCostOverviewStateKey } from 'store/breakdown/costOverview/azureCostOverview';
import { gcpCostOverviewReducer, gcpCostOverviewStateKey } from 'store/breakdown/costOverview/gcpCostOverview';
import { ibmCostOverviewReducer, ibmCostOverviewStateKey } from 'store/breakdown/costOverview/ibmCostOverview';
import { ocpCostOverviewReducer, ocpCostOverviewStateKey } from 'store/breakdown/costOverview/ocpCostOverview';
import { awsHistoricalDataReducer, awsHistoricalDataStateKey } from 'store/breakdown/historicalData/awsHistoricalData';
import {
  azureHistoricalDataReducer,
  azureHistoricalDataStateKey,
} from 'store/breakdown/historicalData/azureHistoricalData';
import { gcpHistoricalDataReducer, gcpHistoricalDataStateKey } from 'store/breakdown/historicalData/gcpHistoricalData';
import { ibmHistoricalDataReducer, ibmHistoricalDataStateKey } from 'store/breakdown/historicalData/ibmHistoricalData';
import { ocpHistoricalDataReducer, ocpHistoricalDataStateKey } from 'store/breakdown/historicalData/ocpHistoricalData';
import { costModelsReducer, costModelsStateKey } from 'store/costModels';
import { awsDashboardReducer, awsDashboardStateKey } from 'store/dashboard/awsDashboard';
import { awsOcpDashboardReducer, awsOcpDashboardStateKey } from 'store/dashboard/awsOcpDashboard';
import { azureDashboardReducer, azureDashboardStateKey } from 'store/dashboard/azureDashboard';
import { azureOcpDashboardReducer, azureOcpDashboardStateKey } from 'store/dashboard/azureOcpDashboard';
import { gcpDashboardReducer, gcpDashboardStateKey } from 'store/dashboard/gcpDashboard';
import { gcpOcpDashboardReducer, gcpOcpDashboardStateKey } from 'store/dashboard/gcpOcpDashboard';
import { ibmDashboardReducer, ibmDashboardStateKey } from 'store/dashboard/ibmDashboard';
import { ocpCloudDashboardReducer, ocpCloudDashboardStateKey } from 'store/dashboard/ocpCloudDashboard';
import { ocpDashboardReducer, ocpDashboardStateKey } from 'store/dashboard/ocpDashboard';
import {
  ocpInfrastructureDashboardReducer,
  ocpInfrastructureDashboardStateKey,
} from 'store/dashboard/ocpInfrastructureDashboard';
import {
  ocpSupplementaryDashboardReducer,
  ocpSupplementaryDashboardStateKey,
} from 'store/dashboard/ocpSupplementaryDashboard';
import { ocpUsageDashboardReducer, ocpUsageDashboardStateKey } from 'store/dashboard/ocpUsageDashboard';
import { exportReducer, exportStateKey } from 'store/exports';
import { forecastReducer, forecastStateKey } from 'store/forecasts';
import { orgReducer, orgStateKey } from 'store/orgs';
import { priceListReducer, priceListStateKey } from 'store/priceList';
import { reportReducer, reportStateKey } from 'store/reports';
import { resourceReducer, resourceStateKey } from 'store/resources';
import { sourcesReducer, sourcesStateKey } from 'store/sourceSettings';
import { tagReducer, tagStateKey } from 'store/tags';
import { StateType } from 'typesafe-actions';

import { metricsReducer, metricsStateKey } from './metrics';
import { providersReducer, providersStateKey } from './providers';
import { rbacReducer, rbacStateKey } from './rbac';
import { uiReducer, uiStateKey } from './ui';
import { userAccessReducer, userAccessStateKey } from './userAccess';

export type RootState = StateType<typeof rootReducer>;

export const rootReducer = combineReducers({
  [awsOcpDashboardStateKey]: awsOcpDashboardReducer,
  [awsCostOverviewStateKey]: awsCostOverviewReducer,
  [awsDashboardStateKey]: awsDashboardReducer,
  [awsHistoricalDataStateKey]: awsHistoricalDataReducer,
  [azureOcpDashboardStateKey]: azureOcpDashboardReducer,
  [azureCostOverviewStateKey]: azureCostOverviewReducer,
  [azureDashboardStateKey]: azureDashboardReducer,
  [azureHistoricalDataStateKey]: azureHistoricalDataReducer,
  [costModelsStateKey]: costModelsReducer,
  [exportStateKey]: exportReducer,
  [gcpCostOverviewStateKey]: gcpCostOverviewReducer,
  [gcpDashboardStateKey]: gcpDashboardReducer,
  [gcpOcpDashboardStateKey]: gcpOcpDashboardReducer,
  [gcpHistoricalDataStateKey]: gcpHistoricalDataReducer,
  [ibmCostOverviewStateKey]: ibmCostOverviewReducer,
  [ibmDashboardStateKey]: ibmDashboardReducer,
  [ibmHistoricalDataStateKey]: ibmHistoricalDataReducer,
  [metricsStateKey]: metricsReducer,
  [ocpCostOverviewStateKey]: ocpCostOverviewReducer,
  [ocpDashboardStateKey]: ocpDashboardReducer,
  [ocpCloudDashboardStateKey]: ocpCloudDashboardReducer,
  [ocpHistoricalDataStateKey]: ocpHistoricalDataReducer,
  [ocpInfrastructureDashboardStateKey]: ocpInfrastructureDashboardReducer,
  [ocpSupplementaryDashboardStateKey]: ocpSupplementaryDashboardReducer,
  [ocpUsageDashboardStateKey]: ocpUsageDashboardReducer,
  [orgStateKey]: orgReducer,
  [priceListStateKey]: priceListReducer,
  [providersStateKey]: providersReducer,
  [rbacStateKey]: rbacReducer,
  [reportStateKey]: reportReducer,
  [forecastStateKey]: forecastReducer,
  [resourceStateKey]: resourceReducer,
  [sourcesStateKey]: sourcesReducer,
  [tagStateKey]: tagReducer,
  [uiStateKey]: uiReducer,
  [userAccessStateKey]: userAccessReducer,
  notifications,
});
