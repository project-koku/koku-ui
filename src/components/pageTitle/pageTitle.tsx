import messages from 'locales/messages';
import React from 'react';
import { injectIntl, WrappedComponentProps } from 'react-intl';
import { RouteComponentProps, withRouter } from 'react-router';
import { paths, routes } from 'routes';

interface PageTitleOwnProps extends RouteComponentProps<void> {
  children?: React.ReactNode;
}

interface PageTitleState {
  // TBD...
}

type PageTitleProps = PageTitleOwnProps & WrappedComponentProps;

class PageTitleBase extends React.Component<PageTitleProps> {
  protected defaultState: PageTitleState = {
    // TBD...
  };
  public state: PageTitleState = { ...this.defaultState };

  private getPath() {
    const { location }: any = this.props;

    // cost models may include :uuid
    const _pathname =
      location.pathname && location.pathname.startsWith(paths.costModels) ? paths.costModels : location.pathname;
    const currRoute = routes.find(({ path }) => path === _pathname);

    return currRoute ? currRoute.path : undefined;
  }

  private getPageTitle() {
    const path = this.getPath();

    switch (path) {
      case paths.explorer:
        return messages.PageTitleExplorer;
      case paths.overview:
        return messages.PageTitleOverview;
      case paths.awsDetails:
      case paths.awsDetailsBreakdown:
        return messages.PageTitleAWS;
      case paths.azureDetails:
      case paths.azureDetailsBreakdown:
        return messages.PageTitleAzure;
      case paths.costModels:
        return messages.PageTitleCostModels;
      case paths.gcpDetails:
      case paths.gcpDetailsBreakdown:
        return messages.PageTitleGCP;
      case paths.ibmDetails:
      case paths.ibmDetailsBreakdown:
        return messages.PageTitleIBM;
      case paths.ocpDetails:
      case paths.ocpDetailsBreakdown:
        return messages.PageTitleOCP;
      default:
        return messages.PageTitleDefault;
    }
  }

  public render() {
    const { children = null, intl } = this.props;

    // Set page title
    document.title = intl.formatMessage(this.getPageTitle());

    return children;
  }
}

const PageTitle = injectIntl(withRouter(PageTitleBase));

export { PageTitle };
