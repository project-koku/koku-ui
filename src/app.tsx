import { I18nProvider } from 'components/i18n';
import { getLocale } from 'components/i18n/localeEnv';
import Maintenance from 'pages/state/maintenance';
import React from 'react';
import { IntlProvider } from 'react-intl';
import { connect } from 'react-redux';
import { RouteComponentProps, withRouter } from 'react-router-dom';
import { compose } from 'redux';
import { createMapStateToProps } from 'store/common';

import { Routes, routes } from './routes';

export interface AppOwnProps {
  basename: string;
}

interface AppStateProps {}

interface AppDispatchProps {
  history: any;
}

interface AppState {
  locale: string;
  maintenanceMode: boolean;
}

type AppProps = AppOwnProps & AppStateProps & AppDispatchProps & RouteComponentProps<void>;

export class App extends React.Component<AppProps, AppState> {
  public appNav: any;

  // Todo: Will Insights provide a flag to enable maintenance mode?
  // https://docs.google.com/document/d/1VLs7vFczWUzyIpH6EUsTEpJugDsjeuh4a_azs6IJbC0/edit#
  public state: AppState = { locale: 'en', maintenanceMode: false };

  public componentDidMount() {
    const { history, location } = this.props;

    insights.chrome.init();
    insights.chrome.identifyApp('cost-management');

    if (location && location.pathname) {
      insights.chrome.appAction(location.pathname);
    }

    this.appNav = insights.chrome.on('APP_NAVIGATION', event => {
      let currRoute = routes.find(({ path }) => path.includes(event.navId));
      /**
       * Condition is required until new nav changes are propagated to each environment.
       * Eventually will be avaiable as a hook value to replace event listening.
       */
      if (!currRoute && typeof event?.domEvent?.href === 'string') {
        const appPathname = event?.domEvent?.href.replace(this.props.basename.replace(/^\/beta\//, '/'), '/');
        currRoute = routes.find(({ path }) => path.includes(appPathname));
      }

      if (event.domEvent && currRoute) {
        history.push(currRoute.path);
      }
    });
  }

  public componentDidUpdate(prevProps: AppProps) {
    const { location } = this.props;

    if (location && location.pathname !== prevProps.location.pathname) {
      window.scrollTo(0, 0);
      insights.chrome.appAction(location.pathname);
    }
  }

  public componentWillUnmount() {
    this.appNav();
    insights.chrome.appAction(undefined);
  }

  public render() {
    const { maintenanceMode } = this.state;
    const route = maintenanceMode ? <Maintenance /> : <Routes />;
    const local = getLocale();

    return (
      <IntlProvider locale={local}>
        <I18nProvider locale={this.state.locale}>{route}</I18nProvider>
      </IntlProvider>
    );
  }
}

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const mapStateToProps = createMapStateToProps<AppOwnProps, AppStateProps>((state, props) => {
  return {};
});

const mapDispatchToProps: AppDispatchProps = { history };

export default compose<React.ComponentType<AppOwnProps>>(withRouter, connect(mapStateToProps, mapDispatchToProps))(App);
