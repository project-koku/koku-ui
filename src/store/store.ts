import { notificationsMiddleware } from '@redhat-cloud-services/frontend-components-notifications';
import axios from 'axios';
import { applyMiddleware, compose, createStore, DeepPartial } from 'redux';
import thunk from 'redux-thunk';

import { rootReducer, RootState } from './rootReducer';

// eslint-disable-next-line @typescript-eslint/no-unused-vars
declare global {
  interface Window {
    __REDUX_DEVTOOLS_EXTENSION_COMPOSE__(options: any): typeof compose;
  }
}

const composeEnhancers =
  typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({ serialize: true })
    : compose;

export const middlewares = [thunk, notificationsMiddleware()];

export function configureStore(initialState: DeepPartial<RootState>) {
  const enhancer = composeEnhancers(applyMiddleware(...middlewares));
  const store = createStore(rootReducer, initialState as any, enhancer);

  axios.interceptors.response.use(null, error => {
    return Promise.reject(error);
  });

  return store;
}
