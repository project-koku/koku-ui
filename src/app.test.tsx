import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { IntlProvider } from '@redhat-cloud-services/frontend-components-translations';
import { createStore } from 'redux';
import { App, AppOwnProps } from './app';
import { FetchStatus } from './store/common';
import { RootState } from '../src/store';
import { rootReducer } from '../src/store/rootReducer';
import messages from '../locales/data.json';
import { injectIntl, IntlCache, IntlConfig } from 'react-intl';
import { withRouter } from 'react-router-dom';
import { PageTitle } from '../src/components/pageTitle/pageTitle'

//TODO: enzyme - these tests don't really do anything, perhaps make them useful or remove?

const props: AppOwnProps = {
  fetchProviders: jest.fn(),
  awsProviders: null,
  awsProvidersFetchStatus: FetchStatus.none,
  awsProvidersQueryString: '',
  history: {
    push: jest.fn(),
    listen: jest.fn(),
  },
  location: null,
  ocpProviders: null,
  ocpProvidersFetchStatus: FetchStatus.none,
  ocpProvidersQueryString: '',
} as any;

jest.mock('components/pageTitle/pageTitle', () => {
  return jest.fn(() => PageTitle)
});

xtest('renders login if isLoggedIn is false', () => {
  const state: Partial<RootState> = {};
  const store = createStore(rootReducer, state);
  const history = createMemoryHistory();
  const cache = {} as IntlCache;
  const prevConfig = {} as IntlConfig;

  const view = render(
    <IntlProvider locale={'en'} messages={messages['en']} cache={cache} prevConfig={prevConfig}>
      <Provider store={store}>
        <Router history={history}>
          <App {...props} />
        </Router>
      </Provider>
    </IntlProvider>);
  expect(view.container).toMatchSnapshot();
});