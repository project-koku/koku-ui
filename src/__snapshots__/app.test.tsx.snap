// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders login if isLoggedIn is false 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <I18nProvider
    locale="en"
  >
    <Routes />
  </I18nProvider>
</IntlProvider>
`;

exports[`renders page layout 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={Object {}}
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <I18nProvider
    locale="en"
  >
    <Routes />
  </I18nProvider>
</IntlProvider>
`;
